uTensor/util/uTensor_util.hpp

```diff
@@ -58,9 +57,57 @@ void utensor_exit(void);
 typedef std::vector<uint32_t> Shape;
 
 void printVector(std::vector<uint32_t> vec);
-
-// little endian to big endian
-uint32_t htonl(uint32_t& val);
+#ifdef _WIN32
+   //define something for Windows (32-bit and 64-bit, this part is common)
+   #ifdef _WIN64
+      //define something for Windows (64-bit only)
+   #else
+      //define something for Windows (32-bit only)
+   #endif
+#elif __APPLE__
+    #include "TargetConditionals.h"
+    #if TARGET_IPHONE_SIMULATOR
+         // iOS Simulator
+    #elif TARGET_OS_IPHONE
+        // iOS device
+    #elif TARGET_OS_MAC
+        // Other kinds of Mac OS
+        #include <arpa/inet.h>
+        #include <sys/stat.h>
+        #include <dirent.h>
+
+
+    #else
+    #   error "Unknown Apple platform"
+    #endif
+#elif __linux__
+    // linux
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <dirent.h>
+#elif __unix__ // all unices not caught above
+    // Unix
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <dirent.h>
+#elif defined(_POSIX_VERSION)
+    // POSIX
+#else
+//#   error "Unknown compiler"
+    // little endian to big endian
+    //uint32_t htonl(uint32_t& val);
+    inline uint32_t htonl(uint32_t& val) {
```

Comment:
Won't `inline` introduces larger code size?
Why not put the definition in the `.cpp` file and guard it with the same preprocessor guards?

Reply:
If the code size is not a big issue, I'm good with `inline`.

Reply:
I believe this is meant to run on host machines, not MCUs, as it is meant for CI.

Reply:
ah...  I see what you mean now.  I thought the inline function was for non-MCU targets.

Reply:
Yeah I thought about that. Probably should make the copy in the cpp file